packages/providers/mgt-msal2-provider/src/Msal2Provider.ts

```diff
@@ -322,6 +362,7 @@ export class Msal2Provider extends IProvider {
    * @memberof Msal2Provider
    */
   public setActiveAccount(user: IProviderAccount) {
+    Providers.unsetCacheId();
```

Comment:
It looks like you are interacting with the cache system directly from a provider implementation. I wonder if there is a way to handle these cache function in a generic way for all providers. If we can add it to the BaseProvider, then implementations won't have to worry about it.

What do you think?

Reply:
Oh yeah I did think of doing that, that's a good idea.